packages/datetime/src/dateRangeInput.tsx

```diff
@@ -111,36 +113,40 @@ export interface IDateRangeInputProps extends IDatePickerBaseProps, IProps {
 
 export interface IDateRangeInputState {
     isOpen?: boolean;
+    preferredBoundaryToModify?: DateRangeBoundary;
+    mostRecentlyFocusedField?: DateRangeBoundary;
 
     isStartInputFocused?: boolean;
     isEndInputFocused?: boolean;
 
     startInputString?: string;
     endInputString?: string;
 
+    startHoverString?: string;
+    endHoverString?: string;
+
     selectedEnd?: moment.Moment;
     selectedStart?: moment.Moment;
+
+    wasLastFocusChangeDueToHover?: boolean;
 };
 
 interface IStateKeysAndValuesObject {
     keys: {
+        hoverString: "startHoverString" | "endHoverString";
         inputString: "startInputString" | "endInputString";
         isInputFocused: "isStartInputFocused" | "isEndInputFocused";
         selectedValue: "selectedStart" | "selectedEnd";
     };
```

Comment:
might it make sense to refactor state to store two instances of this object? `start` and `end`? then you wouldn't have to do all the cooky object creation below. could simply do `this.state[boundary].inputString`.